armi/physics/neutronics/latticePhysics/latticePhysicsWriter.py

```diff
@@ -241,11 +242,19 @@ def _getAllNuclidesByCategory(self, component=None):
         # on the components will already contain all the nuclides required to be
         # modeled by the lattice physics writer. Otherwise, assume that `allNuclidesInProblem`
         # should be modeled.
-        nuclides = (
-            sorted(objNuclides)
-            if self.explicitFissionProducts
-            else self.r.blueprints.allNuclidesInProblem
-        )
+        if self.explicitFissionProducts:
+            # If detailed axial expansion is active, mapping between blocks occurs on uniform mesh
+            # and this can cause blocks to have isotopes that they don't have cross sections for.
+            # Fix this by adding all isotopes so they are present it lattice physics.
+            if self.cs[CONF_DETAILED_AXIAL_EXPANSION]:
+                nuclides = self.r.blueprints.allNuclidesInProblem
+            else:
+                nuclides = sorted(objNuclides)
+        else:
+            nuclides = self.r.blueprints.allNuclidesInProblem
+
+        nuclides = nuclides.union(self.r.core.blueprints.nucsToForceInXsGen)
```

Comment:
I need some explanation of why this `nucsToForceInXsGen` is required still. I think this should be equivalent to just using `self.r.blueprints.allNuclidesInProblem`, so basically we would just have:

`nuclides = self.r.blueprints.allNuclidesInProblem` rather than having a branch for `nuclides = sorted(objNuclides)`, right?

Reply:
no its not possible with 1D control cross section generation, putting everything in the problem in there causes it to take way too long and use more memory than VDI has available. We need a way to disriminate between something the original problem request and `allNuclidesInProblem` which has 400+ isotopes from end 7.0